### ISSUE TITLE
Generalise to non-scalar task outputs: object detection (fasterrcnn) + groundwork for segmentation/structured tasks

### ISSUE THREAD

**Issue by stanlrt** [CAIIVS/raitap#146]
## Summary

Add object-detection support (`fasterrcnn_resnet50_fpn_v2` from the official Aufgabenstellung) by **generalising the task layer**, not patching a single model. Cascades into reusable groundwork for segmentation, structured prediction, and any non-scalar-output task family.

## Motivation

The Aufgabenstellung lists `fasterrcnn_resnet50_fpn_v2` as a required torchvision model. Current pipeline only handles classification:

- `TorchBackend.__call__` (`src/raitap/models/backend.py`) returns a single tensor; detection models return `list[dict[str, Tensor]]` with variable N boxes per image.
- Explainers (Captum, SHAP, Grad-CAM) require a **scalar target**; detection has no single scalar (objectness vs. class score vs. bbox regression — and N of each).
- Empirical attacks (foolbox, torchattacks) optimise `loss(model(x+δ), y_true)` against a classification head — no equivalent for box-set outputs.
- Visualisers (`CaptumImageVisualiser`, etc.) emit one heatmap per sample; detection needs per-box overlays + bbox rendering.

A per-model fix is throwaway. We need a task-level abstraction.

### 1. Target selector (task-generic)
```python
class DetectionTarget:
    """Reduce list-of-dicts model output → scalar for any detector."""
    def __init__(self, box_idx: int, mode: Literal["class_score", "objectness", "bbox_l2"]): ...
    def __call__(self, model_out) -> Tensor: ...
```
Plug into a thin `nn.Module` wrapper exposing classification-shaped API. Existing explainers run **unchanged** on the wrapper. Same pattern reusable for segmentation (`SegmentationTarget(pixel_xy, class)`), token tasks (`TokenTarget(idx, class)`), regression, etc.

### 2. Adversarial loss (task-generic)
```python
class DetectionAdversarialLoss:  # DAG-style: suppress all true boxes
    def __call__(self, pred, true) -> Tensor: ...
```
Drop-in for PGD/FGSM regardless of detector. Mirrors `SegPGD` from literature.

### 3. Visualiser (task-generic)
`DetectionImageVisualiser` — draws boxes + per-box attribution panels + class/score labels. Works for any detector that fits the standard output schema.

## Taxonomy extensions

In `transparency/contracts.py` and `robustness/contracts.py`:

- Extend `ExplanationOutputSpace`: add `DETECTION_BOXES`, `SEGMENTATION_MASK`, `BBOX_REGRESSION`.
- New `TaskKind` enum: `CLASSIFICATION | DETECTION | SEGMENTATION | SEQ2SEQ | REGRESSION`.
- Each adapter declares `supported_tasks: ClassVar[frozenset[TaskKind]]` — `SemanticallyDescribable` registry routes automatically, no per-model glue.
- `infer_output_space` extended to detect torchvision detection models from `model.backend.as_model_for_explanation()`.

## Bonuses unlocked after this lands

- Segmentation: swap target selector to `mask_logit[c, y, x]`, swap visualiser → done.
- Token classification / NER: target = `logits[token_idx, class]`, reuse wrapper.
- Any structured output: reuses the wrapper-to-scalar pattern.

Marginal cost of each subsequent detector / segmenter: ~half a day (config + smoke test).

## Scope

**In scope (this issue):**
- `TaskKind` enum + `supported_tasks` ClassVar across adapters
- Detection output adapter on `ModelBackend` + detection-aware collate
- `DetectionTarget` + scalar-wrapper `nn.Module`
- `DetectionImageVisualiser`
- `DetectionAdversarialLoss` + integration in foolbox/torchattacks adapters
- `fasterrcnn_resnet50_fpn_v2` model config
- ONNX export path validated
- E2E test: detection model → explanation → attack → PDF section
- Hydra config: `configs/model/fasterrcnn.yaml`, `configs/data/<det dataset>.yaml`
- Docs page: "Adding a new task family"

**Out of scope (follow-up issues):**
- Segmentation task family (separate issue, reuses this groundwork)
- Mask R-CNN (separate; depends on this + segmentation)
- DETR / transformer detectors (separate; depends on this)
- Detection-specific formal verification (Marabou not designed for this)

## Acceptance criteria

- [ ] `TaskKind` enum + `supported_tasks` rolled out across all explainer / visualiser / assessor adapters
- [ ] `ExplanationOutputSpace.DETECTION_BOXES` added; `infer_output_space` detects torchvision detection models
- [ ] `DetectionTarget` + scalar wrapper covered by unit tests
- [ ] `DetectionImageVisualiser` produces per-box heatmap + bbox overlay PDF panel
- [ ] At least one empirical attack (PGD or FGSM) succeeds against `fasterrcnn_resnet50_fpn_v2` in CI
- [ ] E2E test: load model → run SHAP/Captum/Grad-CAM → run attack → emit PDF section → log to MLflow
- [ ] Public docs page documents the task-family extension pattern (so future contributors can add segmentation/seq2seq)

## Effort estimate

~2–3 weeks single engineer:

| Layer | Days |
|---|---|
| `TaskKind` + `supported_tasks` + routing | 1–2 |
| Detection output adapter + collate | 1–2 |
| `DetectionTarget` + scalar wrapper + explainer wiring | 3–7 |
| `DetectionImageVisualiser` + PDF section | 2–3 |
| `DetectionAdversarialLoss` + attack wiring | 3–5 |
| Configs, e2e tests, docs | 1–2 |

## References

- DAG (Dense Adversary Generation) — detection-specific adversarial attack
- Daedalus — bbox-spawning attack
- `SegPGD` — task-loss-adapted PGD (template for detection version)
- torchvision detection model API: `torchvision.models.detection`
